### PR TITLE
Add resource count to plugin and project generator, also add resource generation pipeline

### PR DESCRIPTION
**Pull request by Temennigru**
Adding resource counting to the model, plugin and project generator to be used in the future resource generator

Test: None

### CODE REVIEW

File: code/plugin/src/main/kotlin/com/android/gradle/replicator/codegen/plugin/GenerateResources.kt [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+import org.gradle.api.file.DirectoryProperty
+import org.gradle.api.file.RegularFileProperty
+import org.gradle.api.provider.Property
+import org.gradle.api.tasks.*
+import kotlin.random.Random
+
+abstract class GenerateResources: DefaultTask() {
+
+    @get:InputFile
+    abstract val resourceGenParamFile: RegularFileProperty
```
Comment (jdochez) [inline code/plugin/src/main/kotlin/com/android/gradle/replicator/codegen/plugin/GenerateResources.kt:29]: parameterFile (well same name as for the codegen task really)

File: model/src/main/kotlin/com/android/gradle/replicator/model/internal/DefaultAndroidResourcesInfo.kt [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+) : AndroidResourcesInfo {
+    override val asMap: MutableMap<String, MutableMap<String, MutableMap<String, Int>>> =
+        fileCount.folders.mapValues { folder ->
+                folder.value.qualifiers.mapValues { qualifier ->
+                            qualifier.value.extensions
+                } as MutableMap
+        } as MutableMap
+}
+
+/* Resource division is done in the following way:
```
Comment (jdochez) [inline model/src/main/kotlin/com/android/gradle/replicator/model/internal/DefaultAndroidResourcesInfo.kt:53]: remove ?

File: code/codegen/src/main/kotlin/com/android/gradle/replicator/resgen/Main.kt [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+                    lastKey = s
+                }
+                else this[lastKey]?.add(s)
+            }
+        }
+
+        val argumentsBuilder = ResourceGenerationParameters.Builder()
+        parseArguments(parsedArguments, argumentsBuilder)
+
+        val androidOutputFolder = File(checkNotNull(parsedArguments["-ao"]).first())
```
[the remark this answers is not in the document]
Comment (Temennigru) [reply code/codegen/src/main/kotlin/com/android/gradle/replicator/resgen/Main.kt:57]: I consolidated args parsing into a class to make it more consistent. Take a look at the latest changeset

File: code/codegen/build.gradle.kts [hunk 4]
```diff
@@ -51,8 +56,10 @@ kotlin {
 dependencies {
     implementation(project(":model"))
 
+    compileOnly("org.jetbrains.kotlin:kotlin-gradle-plugin:1.3.72")
```
[the remark this answers is not in the document]
Comment (Temennigru) [reply code/codegen/build.gradle.kts:59]: ack

**Pull request merged**